File: years/2015/day1.ts
// https://adventofcode.com/2015/day/1

import { readFileSync } from 'fs';

const input = readFileSync(__filename.replace('.ts', '-input'), 'utf-8');

// First Star:
let floor = 0;
let firstBasement = null;
let charNum = 0;
for (const char of input) {
  charNum++;
  if (char === '(') floor++;
  else floor--;
  if (floor === -1 && firstBasement === null) firstBasement = charNum;
}
console.log({ floor });

// Second Star:
console.log({ firstBasement });

File: years/2015/day1.rs
// use ferris_says::say; // from the previous step
use std::fs;
use std::path::{Path, PathBuf};

fn input_path(year: u16, day: u16) -> PathBuf {
    return Path::new("tmp").join(format!("{}-{}-input", year, day));
}

fn main() {
    let path = input_path(2015, 1);
    // println!("In file {}", path.to_str().unwrap());
    let contents = fs::read_to_string(path).unwrap();
    // println!("With text: {}", contents);
    let mut floor: i16 = 0;
    let mut first_basement: u16 = 0;
    let mut char_num: u16 = 0;
    for c in contents.chars() {
        char_num += 1;
        if c == '(' {
            floor += 1;
        } else {
            floor -= 1;
        }
        if floor == -1 && first_basement == 0 {
            first_basement = char_num;
        }
    }

    println!("Floor: {}", floor);
    println!("First basement: {}", first_basement);
}
